### ISSUE TITLE
Found a bug with the shuffle in the music app

### ISSUE THREAD

**Issue by JKOBPS** [freeCodeCamp/freeCodeCamp#56095]
### Describe the Issue

There is a bug on the music app, when you delete all the songs and click shuffle, the reset playlist button dissappear, it cause to dont have acces to do anything else with the app, the only solution is to refresh the web.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-string-and-array-methods-by-building-a-music-player/step-99

### Steps to Reproduce

1. Delete all the songs.
2. Click on Shuffle.
3. There is no way to reset the list by using the app.

### Expected behavior

I expected to still see the reset button in UI, not to delete it.

### Screenshots

![BugMusicApp](https://github.com/user-attachments/assets/4703f015-b523-449a-9478-4ff7ceef9289)
As you can see, step 99, and doing the bug the reset button dissappear.

### System

- Device: [Alls]
- Browser: [Alls]

### Additional context

Would be good to add another step by resolving this bug :D

**Comment by gikf** [comment]
You are right. The _Reset Playlist_ button is added, to the songs list, by `deleteSong` when deleting the last song. When `shuffle` is used, songs list calls `renderSongs` which clears the button.

I think better place for adding _Reset Playlist_ button would be `renderSongs` function. Alternatively _shuffle_ could return early if there's no songs.

**Comment by itsjatinnagar** [comment]
Hi @gikf 
I am working on the **fix** for this issue.
The only thing is, I am not sure in which *Step* of the Curriculum the code snippet should be introduced.

**Comment by iamkaranroushan** [comment]
We just have to write the `resetButton` logic separately and use it in both the `shuffle` and `DeleteSong` to make sure that `resetButton` is correctly rendered when there is no songs in the playlist.

**Comment by harshaguntur** [comment]
hey can i work on this ?

**Comment by ilenia-magoni** [comment]
@prabhakaryadav2003 adding it at the end is fine, something like "there is one more bug...", and if there is a step that say "there is one last bug" earlier it can be changed

**Comment by NikaJamaspishvili** [comment]
Yeah the bug is really harsh, we should probably modify the lambda function change it to lambda ssr. this will solve the problem and it can also make it possible to integrate other hr capabilities such as: wrc and trt managment.
